libr/core/linux_heap_jemalloc.c

```diff
@@ -372,6 +372,89 @@ static void GH(jemalloc_get_bins)(RCore *core, const char *input) {
 	}
 }
 
+static void GH(jemalloc_get_runs)(RCore *core, const char *input) {
+	switch (input[0]) {
+	case ' ':
+		{
+			int i, j;
+			GHT addr = UT64_MAX, size = UT64_MAX, mapbits = UT64_MAX, flags = UT64_MAX, mask = UT64_MAX, chunk = UT64_MAX, misc_offset=UT64_MAX;
+			GHT map = UT64_MAX, map_misc_offset = UT64_MAX, chunksize_mask = UT64_MAX, npages = UT64_MAX, map_bias = UT64_MAX;
+			arena_chunk_t *c = R_NEW0 (arena_chunk_t);
+
+			input += 1;
+			chunk = strstr (input, "0x") ? (GHT)strtol (input, NULL, 0) : (GHT)strtol (input, NULL, 16);
+
+			if (GH(r_resolve_jemalloc)(core, "je_chunk_npages", &npages)) {
+				r_core_read_at (core, npages, (ut8*)&npages, sizeof (GHT));
+			} else {
+				eprintf ("Error resolving je_chunk_npages\n");
+				return;
+			}
+
+			if (GH(r_resolve_jemalloc)(core, "je_chunksize_mask", &chunksize_mask)) {
+				r_core_read_at (core, chunksize_mask, (ut8*)&chunksize_mask, sizeof (GHT));
+			} else {
+				eprintf ("Error resolving je_chunksize_mask\n");
+				return;
+			}
+
+			if (GH(r_resolve_jemalloc)(core, "je_map_bias", &map_bias)) {
+				r_core_read_at (core, map_bias, (ut8*)&map_bias, sizeof (GHT));
+			} else {
+				eprintf ("Error resolving je_map_bias");
+			}
+
+			if (GH(r_resolve_jemalloc)(core, "je_map_misc_offset", &map_misc_offset)) {
+				r_core_read_at (core, map_misc_offset, (ut8*)&map_misc_offset, sizeof (GHT));
+			} else {
+				eprintf ("Error resolving je_map_misc_offset");
+			}
+
+			printf("map_misc_offset %x\n", map_misc_offset);
+
+			r_core_read_at (core, chunk, (ut8 *)c, sizeof (arena_chunk_t));
+			mapbits = *(GHT *)&c->map_bits;
+			eprintf ("map_bits: 0x%"PFMTx"\n", mapbits);
+
+			uint32_t offset = r_offsetof(arena_chunk_t, map_bits);
+
+			arena_chunk_map_bits_t *dwords = (size_t*)calloc(sizeof(arena_chunk_map_bits_t), npages);
+			r_core_read_at (core, chunk+offset, (ut8*)dwords, sizeof(arena_chunk_map_bits_t) * npages);
+			printf("map_bits @ 0x%"PFMTx"\n", chunk+offset);
```

Comment:
use PFMT64, and ut64 as much as possible. having GHT will make the code more complicated for formatting strings, etc.